app/views/spree/components/forms/inputs/_text.html.erb

```diff
@@ -7,11 +7,13 @@
   label = false if local_assigns[:label].nil?
   min = false if local_assigns[:min].nil?
   placeholder = "" if local_assigns[:placeholder].nil?
+  small = false if local_assigns[:small].nil?
```

Comment:
I'm not really onboard with this: being small is something that works for our design but someone else could want to have that thing (also applies to button component) bigger. What about moving these rules to CSS only?

Reply:
I'm not sure I'm following. `small` is toggle-able. If the user wants the button to be bigger (normal/default size), just don't pass anything when rendering the button component. 

Also, how would you go about moving the rule to CSS only?

Reply:
I mean that we can replace the `-small` definition from the HTML to the CSS. Something like (this is pseudo-code):

```
.coupon-code {
  .text-input {
    // code that make the button small
  }
}
```

In my perspective, this way, making the button smaller is a responsibility of the CSS only. If someone wants to make it bigger, it's a style/look&feel concern, they should not touch the HTML.

Reply:
The issue here is that if we need the button smaller in another place (we will be needing it in the checkout process), we would need to duplicate the css is several places.

Passing props that control the style (size, color, visibility etc.) is a common pattern in React for example. It avoids a lot of code duplication.